clang/include/clang/CIR/Dialect/IR/CIRAttrs.td

```diff
@@ -264,6 +264,71 @@ def ConstPtrAttr : CIR_Attr<"ConstPtr", "ptr", [TypedAttrInterface]> {
   let hasCustomAssemblyFormat = 1;
 }
 
+//===----------------------------------------------------------------------===//
+// ThreeWayCmpInfoAttr
+//===----------------------------------------------------------------------===//
+
+def ThreeWayCmpOrderingKind_Strong : I32EnumAttrCase<"Strong", 1, "strong">;
+def ThreeWayCmpOrderingKind_Partial : I32EnumAttrCase<"Partial", 2, "partial">;
+
+def ThreeWayCmpOrderingKind : I32EnumAttr<
+  "ThreeWayCmpOrderingKind", "three-way comparison ordering kind",
+  [ThreeWayCmpOrderingKind_Strong, ThreeWayCmpOrderingKind_Partial]
+> {
+  let cppNamespace = "::mlir::cir";
+}
+
+def ThreeWayCmpInfoAttr
+    : CIR_Attr<"ThreeWayCmpInfo", "three_way_cmp_info"> {
+  let summary = "Holds information about a three-way comparison operation";
+  let description = [{
+    The `#three_way_cmp_info` attribute contains information about a three-way
+    comparison operation `cir.three_way_cmp`.
+
+    The `ordering` parameter gives the ordering kind of the three-way comparison
+    operation. It may be either strong ordering or partial ordering.
+
+    Given the two input operands of the three-way comparison operation `lhs` and
+    `rhs`, the `lt`, `eq`, `gt`, and `unordered` parameters gives the result
+    value that should be produced by the three-way comparison operation when the
+    ordering between `lhs` and `rhs` is `lhs < rhs`, `lhs == rhs`, `lhs > rhs`,
+    or neither, respectively.
```

Comment:
Please add a short example of how to use this operation.

Reply:
Updated.